### ISSUE TITLE
Unexpected working of maximize_structured function when using multiple arguments

### ISSUE THREAD

**Issue by suzukikbp** [claesenm/optunity#53]
I would like to optimize features' weight and classifier's parameters at the same time using maximize_structured function. I specified search space as below (F1 corresponds to the first feature. In this case, there are 11 features to optimize their weight), and used it in the maximize_structured function. However, after the optimization of first iteration, arguments changed like below, which just returns 'None' for 'weight' argument and optimizes just  'algorithm' arguments. I expected that all parameters of 'weight' argument (i.e. F1-F11) are optimized as well.  How can I optimize 'weight' at the same time with optimization for classifier?

	self.search=
	{'algorithm': {'SVM': {'kernel': {'linear': {'C': [-5, 2]},
	                                  'poly': {'C': [-5, 2],
	                                           'coef0': [0, 1],
	                                           'degree': [2, 5]},
	                                  'rbf': {'C': [-5, 2], 'gamma': [-5, 2]}}},
	               'naive-bayes': None,
	               'random-forest': {'max_features': [2, 1096],
	                                 'n_estimators': [10, 30]}},
	 'weight': {'F1': [1, 10],
	            'F10': [1, 10],
	            'F11': [1, 10],
	            'F2': [1, 10],
	            'F3': [1, 10],
	            'F4': [1, 10],
	            'F5': [1, 10],
	            'F6': [1, 10],
	            'F7': [1, 10],
	            'F8': [1, 10],
	            'F9': [1, 10]}}

This is the arguments that I got inside the performance function.

	kargs
	{'C': 0.88232421875,
	 'F1': None,
	 'F10': None,
	 'F11': None,
	 'F2': None,
	 'F3': None,
	 'F4': None,
	 'F5': None,
	 'F6': None,
	 'F7': None,
	 'F8': None,
	 'F9': None,
	 'algorithm': 'SVM',
	 'coef0': 0.04931640625,
	 'degree': 3.49853515625,
	 'gamma': None,
	 'kernel': 'poly',
	 'max_features': None,
	 'n_estimators': None}

Here is the code where I use the search space and get the arguments 'kargs'.

    @optunity.cross_validated(x=self.xtrain, y=self.ytrain, num_folds=num_folds, num_iter=num_iter)
    def performance(x_train, y_train, x_test, y_test,**kargs):
        xtrain,xtest=setWeight(xtrain,xtest,**kargs)
        model,_=setModel(**kargs)
        model.fit(x_train, y_train)
        # predict the test set
        if kargs['algorithm'] == 'SVM':
            predictions = model.decision_function(x_test)
        else:
            predictions = model.predict_proba(x_test)[:, 1]
        return optunity.metrics.roc_auc(y_test, predictions)

    optimal_configuration, info, optimal_pars3 = optunity.maximize_structured(performance,search_space=self.search,num_evals=num_evals)

**Comment by claesenm** [comment]
The problem appears to be the way you specify the search space. The following:

    'weight': {'F1': [1, 10],
                ....
                }

creates a choice, rather than a vector. That is, you have created a `weight` choice with 11 options.

The correct search space for your problem would be as follows:

    self.search=
    {'algorithm': {'SVM': {'kernel': {'linear': {'C': [-5, 2]},
                                    'poly': {'C': [-5, 2],
                                            'coef0': [0, 1],
                                            'degree': [2, 5]},
                                    'rbf': {'C': [-5, 2], 'gamma': [-5, 2]}}},
                'naive-bayes': None,
                'random-forest': {'max_features': [2, 1096],
                                    'n_estimators': [10, 30]}},
    'F1': [1, 10], 'F10': [1, 10], 'F11': [1, 10], 'F2': [1, 10],
    'F3': [1, 10], 'F4': [1, 10], 'F5': [1, 10], 'F6': [1, 10],
    'F7': [1, 10], 'F8': [1, 10], 'F9': [1, 10]
    }

That is, each weight (`F1`-`F11`) should be on the same level as `algorithm`.

**Comment by claesenm** [comment]
Update: looks like I found a bug, will fix ASAP.
